crates/wasm-mutate/src/mutators/peephole/eggsy/mod.rs

```diff
@@ -443,79 +463,88 @@ impl Encoder {
                 let (operator, _) = &operators[entry.operator_idx];
-                let (term, addtosymbolsmap) = Encoder::get_simpleterm(&operator)?;
-                Ok((
-                    term.clone(),
-                    if addtosymbolsmap {
-                        vec![(term, stack_entry_index)].into_iter().collect()
-                    } else {
-                        HashMap::new()
-                    },
-                ))
+                match operator {
+                    Operator::I32Const { value } => {
+                        let id = expr.add(Lang::I32Const(*value));
+                        Ok((
+                            id,
+                            HashMap::new(), // No symbol
+                        ))
+                    }
+                    Operator::LocalGet { local_index } => {
+                        let name = format!("?l{}", local_index);
+                        let id = expr.add(Lang::Symbol(name.clone().into()));
+                        let mut smap = HashMap::new();
+                        smap.insert(name, stack_entry_index);
+
+                        Ok((id, smap))
+                    }
+                    _ => Err(crate::Error::UnsupportedType(EitherType::Operator(
+                        format!("Operator {:?} is not supported as symbol", operator),
+                    ))),
+                }
             }
             crate::mutators::peephole::dfg::StackEntryData::Node { operands } => {
                 // This is an operator
                 let (operator, _) = &operators[entry.operator_idx];
-                let (term, _) = Encoder::get_simpleterm(&operator)?;
-                let mut operandterms = Vec::new();
+                let mut subexpressions = Vec::new();
                 let mut smap: HashMap<String, usize> = HashMap::new();
+
                 for operandi in operands {
                     let stack_entry = &dfg.entries[*operandi];
                     let (eterm, symbols) =
-                        Encoder::wasm2eterm(dfg, stack_entry.operator_idx, operators, rnd)?;
-                    operandterms.push(eterm);
-                    smap.extend(symbols.into_iter())
+                        Encoder::wasm2expr(dfg, stack_entry.operator_idx, operators, expr)?;
+                    subexpressions.push(eterm);
+                    smap.extend(symbols.into_iter());
                 }
-                Ok((format!("({} {})", term, operandterms.join(" ")), smap))
+                let operatorid = match operator {
+                    Operator::I32Shl => {
+                        // Check this node has only two child
+                        assert_eq!(subexpressions.len(), 2);
+
+                        Ok(expr.add(Lang::I32Shl([subexpressions[0], subexpressions[1]])))
+                    }
+                    Operator::I32Add => {
+                        // Check this node has only two child
+                        assert_eq!(subexpressions.len(), 2);
```

Comment:
These can probably be `debug_assert_eq!` as they are "internal" implementation assertions and not something we are asserting about user/some-other-crate's input.